### PR TITLE
Add ability to disable/hide Copilot suggestions

### LINKED ISSUE #4333: allow me to toggle copilot settings from the status bar
### Check for existing issues

- [X] Completed

### Describe the feature

currently, it's only possible to **disable** copilot settings from the status bar

for example, if copilot is enabled, I can set `show_copilot_suggestions` to `false` by clicking on "hide suggestions for all files"

Bildschirmfoto 2023-11-23 um 12 14 44

once copilot is disabled, however, I have to manually set `show_copilot_suggestions` to true via `settings.json` to re-enable copilot, as there is no option for that in the status bar menu

Bildschirmfoto 2023-11-23 um 12 15 08

all I can do is disable copilot completely, and "sign in", which does nothing in this state

### PR DESCRIPTION
**Pull request by dhruvkelawala**
Release Notes:

- Added the ability to disable/hide Copilot suggestions ([#4333](https://github.com/zed-industries/zed/issues/4333)).

### Screenshot

Screenshot 2024-03-27 at 22 16 35

### CODE REVIEW

File: crates/copilot_ui/src/copilot_button.rs [hunk 1]
```diff
@@ -159,7 +159,11 @@ impl CopilotButton {
                 menu = menu.entry(
                     format!(
                         "{} Suggestions for {}",
-                        if language_enabled { "Hide" } else { "Show" },
```
Comment (maxdeviant) [inline crates/copilot_ui/src/copilot_button.rs:162]: Is there a reason we're changing the terminology used here?

File: crates/copilot_ui/src/copilot_button.rs [hunk 2]
```diff
@@ -198,14 +202,15 @@ impl CopilotButton {
             }
 
             let globally_enabled = settings.copilot_enabled(None, None);
+            let fs_clone_for_first_use = fs.clone();
```
Comment (maxdeviant) [inline crates/copilot_ui/src/copilot_button.rs:205]: I don't think we need this binding named like this. If we need to introduce a new binding we should just call it `fs` and shadow the earlier one.

Comment (dhruvkelawala) [reply crates/copilot_ui/src/copilot_button.rs:205]: Thanks, still working on this PR. I will put it to draft

**Comment by dhruvkelawala** [comment]
@cla-bot check

File: crates/copilot_ui/src/copilot_button.rs (hunk 1 above)
Comment (dhruvkelawala) [reply crates/copilot_ui/src/copilot_button.rs:162]: I think the "Enable" / "Disable" and "Hide" is a better terminology. The current one, where "Disable" hides the copilot from status bar and "Hide from all files" disables it seems counter-intuitive... Happy to change it back if the team think something else is better.

**Comment by maxdeviant** [comment]
We're going to be revisiting this area of the UI as we add support for other completion providers (like Supermaven).

**Pull request closed without merging**